facebook/react-native
# Android Fix for 9145: No longer hard code build port

## Summary

### Problem

According to https://github.com/facebook/react-native/issues/9145, the `--port` setting is not respected when executing `react-native run-android`. The templates that report things like what port the dev server runs on are hard coded as well.

### Solution

This commit replaces the hardcoded instances of port 8081 on Android with a build configuration property. This allows setting of the port React Native Android connects to for the local build server.

For this change to work, there must also be an update to the react native CLI to pass along this setting:

https://github.com/react-native-community/react-native-cli/compare/master...nhunzaker:9145-android-no-port-hardcode-cli

To avoid some noise on their end, I figured I wouldn't submit a PR until it's this approach is deemed workable.

## Changelog

[Android][fixed] - `react-native run-android --port <x>` correctly connects to dev server and related error messages display the correct port

## Test Plan

Preparing a test case for this is pretty involved, but:

1. Clone https://github.com/nhunzaker/react-native-cli/tree/9145-android-no-port-hardcode-cli
2. Install dependencies for this project
4. Clone https://github.com/nhunzaker/react-native/tree/9145-android-no-port-hardcode
5. Install dependencies for this project
6. Create a new RN project
7. Link this branch to the new project following [The Build From Source Instructions](https://facebook.github.io/react-native/docs/building-from-source)

(for future PR's, I'd love figure out a better flow for reproducing fixes)

Then from the new project, execute:

```
yarn start --port 3000
```

And in another terminal window:

```
node ../react-native-cli/packages/cli/ run-android --port 3000
```

You can also do this from gradle via:

```
./gradlew :app:installDebug -PdebugHostPort=3001
```

![image](https://user-images.githubusercontent.com/590904/53294484-e224d000-379c-11e9-82bb-f0397e7760c6.png)

Error messages are also updated, so you'll see the correct port if the server is down:

![image](https://user-images.githubusercontent.com/590904/53294508-59f2fa80-379d-11e9-857b-32ad8aeaa50a.png)
